cgo/lapack.go

```diff
@@ -413,6 +425,63 @@ func (impl Implementation) Dlarfx(side blas.Side, m, n int, v []float64, tau flo
 	lapacke.Dlarfx(side, m, n, v, tau, c, ldc, work)
 }
 
+// Dlascl multiplies an m×n matrix by the scalar cto/cfrom.
+//
+// cfrom must not be zero, and cto and cfrom must not be NaN, otherwise Dlascl
+// will panic.
+//
+// Dlascl is an internal routine. It is exported for testing purposes.
+func (impl Implementation) Dlascl(kind lapack.MatrixType, kl, ku int, cfrom, cto float64, m, n int, a []float64, lda int) {
+	checkMatrix(m, n, a, lda)
+	if cfrom == 0 {
+		panic(zeroDiv)
+	}
+	if math.IsNaN(cfrom) || math.IsNaN(cto) {
+		panic(nanScale)
+	}
+	lapacke.Dlascl(byte(kind), kl, ku, cfrom, cto, m, n, a, lda)
+}
+
+// Dlaset sets the off-diagonal elements of A to alpha, and the diagonal
```

Comment:
This would require

```diff
diff --git a/cgo/lapacke/generate_lapacke.go b/cgo/lapacke/generate_lapacke.go
index 9110c9a..f082d16 100644
--- a/cgo/lapacke/generate_lapacke.go
+++ b/cgo/lapacke/generate_lapacke.go
@@ -421,10 +422,8 @@ func uplo(buf *bytes.Buffer, d binding.Declaration, p binding.Parameter) bool {
                ul = 'U'
        case blas.Lower:
                ul = 'L'
-       case blas.All:
-               ul = 'A'
        default:
-               panic("lapack: illegal triangle")
+               ul = 'A'
        }
 `)
```

in addition to the above changes.